orderer/consensus/hlmirbft/chain.go

```diff
@@ -1056,146 +1091,224 @@ func (c *Chain) triggerCatchup(sn *raftpb.Snapshot) {
 	case <-c.doneC:
 	}
 }
+
 //JIRA FLY2-48 proposed changes: fetch request from request store
-func (c *Chain) fetchRequest(ack *msgs.RequestAck) (*msgs.Request,error){
+func (c *Chain) fetchRequest(ack *msgs.RequestAck) (*msgs.Request, error) {
 
 	reqByte, err := c.Node.ReqStore.GetRequest(ack)
 	if err != nil {
 		return nil, errors.WithMessage(err, "Cannot Fetch Request")
 	}
 	if reqByte == nil {
-		return nil,errors.Errorf("reqstore should have request if we are committing it")
+		return nil, errors.Errorf("reqstore should have request if we are committing it")
 	}
 	reqMsg := &msgs.Request{}
-	err = proto.Unmarshal(reqByte,reqMsg)
+	err = proto.Unmarshal(reqByte, reqMsg)
 	if err != nil {
-		return nil,errors.WithMessage(err, "Cannot Unmarshal Request")
+		return nil, errors.WithMessage(err, "Cannot Unmarshal Request")
 	}
-	return reqMsg,nil
+	return reqMsg, nil
 }
+
 //FLY2-48 proposed changes
 // - convert batches to block and write to the ledger
-func (c *Chain) processBatch( batch *msgs.QEntry) error{
+func (c *Chain) processBatch(batch *msgs.QEntry) error {
 	var envs []*common.Envelope
 	for _, requestAck := range batch.Requests {
-		reqMsg,err := c.fetchRequest(requestAck)
+		reqMsg, err := c.fetchRequest(requestAck)
 		if err != nil {
 			return errors.WithMessage(err, "Cannot fetch request from Request Store")
 		}
-		env,err:= protoutil.UnmarshalEnvelope(reqMsg.Data)
+		env, err := protoutil.UnmarshalEnvelope(reqMsg.Data)
 		if err != nil {
 			return errors.WithMessage(err, "Cannot Unmarshal Request Data")
 		}
 		if c.isConfig(env) {
-			configBlock := c.CreateBlock([]*common.Envelope{env})
-			c.writeConfigBlock(configBlock)
+			reconfig, err := c.getNewReconfiguration(env)
```

Comment:
There are scenarios where `reconfig` should be nil such as a configuration transaction which updates the channel mod policy but doesn't change the ordering node configuration. In such case a config block should be written to the ledger immediately and the loop continued.

For the moment presume that reconfig can be `nil` (I know it can't at the moment but that's fine - it can dealt with in a later ticket) so if `reconfig` is `nil` write a config block and continue the loop.